common/tasks/interleaved_weighted_round_robin.go

```diff
@@ -131,13 +133,25 @@ func (s *InterleavedWeightedRoundRobinScheduler) Stop() {
 
 func (s *InterleavedWeightedRoundRobinScheduler) Submit(
 	task PriorityTask,
-
 ) {
 	channel := s.getOrCreateTaskChannel(s.priorityToWeight[task.GetPriority()])
 	channel.Chan() <- task
 	s.notifyDispatcher()
 }
 
+func (s *InterleavedWeightedRoundRobinScheduler) TrySubmit(
```

Comment:
how do you plan to use this function? 
if try submit fails, then what?

Reply:
Main idea is, compared to `Submit` this function is non-blocking. Since task scheduler is dealing with priority tasks, I don't want the thread to get blocked when submitting a low priority task. Instead, the submission can fail (and the task will be added to the rescheduler) and continue to try submitting the next task which may have a different priority.